### PR TITLE
Fix miner logic

### PR DESCRIPTION
**Pull request by n8maninger**

### CODE REVIEW

File: cmd/walletd/miner.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	for types.BlockID(types.HashBytes(buf)).CmpWork(cs.ChildTarget) < 0 {
+		b.Nonce += factor
+		hashes++
+		binary.LittleEndian.PutUint64(buf[32:], b.Nonce)
+		if time.Since(startBlock) > 10*time.Second {
+			return hashes, false
+		}
+	}
+	return hashes, true
+}
```
Comment (lukechampine) [inline cmd/walletd/miner.go:39]: can use `coreutils.FindBlockNonce` now

Comment (lukechampine) [reply cmd/walletd/miner.go:39]: (maybe even `coreutils.MineBlock`? not sure)

File: cmd/walletd/miner.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+)
+
+func runCPUMiner(c *api.Client, minerAddr types.Address, n int) {
+	log.Println("Started mining into", minerAddr)
+	start := time.Now()
+
+	var last types.ChainIndex
+	var blocksFound int
+	for {
+		if n > 0 && blocksFound >= n {
```
Comment (ChrisSchinnerl) [inline cmd/walletd/miner.go:22]: super nit but `n >= 0` to match the description of the CLI flag.

File: cmd/walletd/miner.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+			break
+		}
+		elapsed := time.Since(start)
+		cs, err := c.ConsensusTipState()
+		check("Couldn't get consensus tip state:", err)
+		if cs.Index == last {
+			fmt.Println("Tip now", cs.Index)
+			last = cs.Index
+		}
```
Comment (ChrisSchinnerl) [inline cmd/walletd/miner.go:31]: why set it if it already matches? Considering that `last` isn't used anywhere else this probably never prints?

**Review by ChrisSchinnerl — COMMENTED** [review]
The fact that the whole method was moved + the lack of PR description makes it quite hard to figure out what the logic is that required fixing ^^

**Comment by n8maninger** [comment]
> The fact that the whole method was moved + the lack of PR description makes it quite hard to figure out what the logic is that required fixing ^^

Don't worry about it 😛

**Review by lukechampine — APPROVED** [review]

**Pull request merged**